### ISSUE TITLE
UC read/write ops in mem_cmd/mem_resp are unchecked

### ISSUE THREAD

**Issue by dpetrisko** [black-parrot/black-parrot#781]
https://github.com/black-parrot/black-parrot/blob/dev/bp_common/src/include/bp_common_bedrock_pkgdef.svh#L45

Do we envision using these? Currently, the L2 does not support uncached / passthrough requests so all requests that end up there are cached regardless. We could work around it by bypassing the L2 entirely...

**Comment by taylor-bsg** [comment]
Seems like we should outline the usage scenarios (what is the motivation
for pass thru) and also the role of cache management instructions. So for
example, if BP is sharing an external HBM bank with an non-coherent
accelerator, then the one option is to bypass the L2 and have no caching,
the other would be to use the L2 and do software managed coherence. This
case actually applies to BigBlade.

**Comment by RKNAGA18** [comment]
hi ,i feel adding strict fatal assertions to mem_cmd and mem_resp to safely trap any Uncached (UC) operations to prevent unsupported L2 behavior. Let me know if this approach works

**Comment by dpetrisko** [comment]
Yeah I’d accept an $error in the l2 cache controller if it receives uncached ops for now. There was actually some work on the l2 to support this feature natively — I’ll have to look if that ever was completed

**Comment by RKNAGA18** [comment]
thanks @dpetrisko  I’ll add the $error checks for uncached operations in the L2 cache controller and open a PR shortly.
